controllers/machineinventory_controller.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	elementalv1 "github.com/rancher/elemental-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	errorutils "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// MachineInventoryReconciler reconciles a MachineInventory object.
+type MachineInventoryReconciler struct {
+	client.Client
+}
+
+// +kubebuilder:rbac:groups=elemental.cattle.io,resources=machineinventories,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=elemental.cattle.io,resources=machineinventories/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;watch;create
+
+func (r *MachineInventoryReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&elementalv1.MachineInventory{}).
+		Owns(&corev1.Secret{}).
+		Watches(
+			&source.Kind{
+				Type: &corev1.Secret{},
+			},
+			&handler.EnqueueRequestForOwner{
+				IsController: true,
+				OwnerType:    &elementalv1.MachineInventory{}}).
+		WithEventFilter(r.ignoreIncrementalStatusUpdate()).
+		Complete(r)
+}
+
+func (r *MachineInventoryReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	mInventory := &elementalv1.MachineInventory{}
+	err := r.Get(ctx, req.NamespacedName, mInventory)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.V(5).Info("Object was not found, not an error")
```

Comment:
🤔 general question about logging... V(5) sets the verbosity level (the higher the less critical). I see we use this all around the new code, and I have some probably trivial questions :
- how is V(5) rendered in the output ?
- why 5?
- I see that we can add to the Info() parameters keys and values... should be handy at some point (later on) to add the obj Namespace/Name or would it be redundant as we get that already?

Reply:
If you run the controller with `-v 5` the logs will be rendered in the same way as any other logs. I chose level 5 because it's commonly used in CAPI for debug logs. Namespace/name of the object should already be present in the logs.